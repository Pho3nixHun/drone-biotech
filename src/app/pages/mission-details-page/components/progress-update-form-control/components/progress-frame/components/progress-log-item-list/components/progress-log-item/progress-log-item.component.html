<div
    class="flex flex-row justify-between items-center px-2 py-1 space-x-4 w-full"
>
    <div class="break-all">
        <div class="font-medium text-sm">
            <ng-content select="h3" />
        </div>
        <div class="text-gray-600 font-light text-sm">
            <ng-content select="p" />
        </div>
    </div>
    <div class="text-gray-600 font-light text-sm">
        <ng-content select="time" />
    </div>
</div>
