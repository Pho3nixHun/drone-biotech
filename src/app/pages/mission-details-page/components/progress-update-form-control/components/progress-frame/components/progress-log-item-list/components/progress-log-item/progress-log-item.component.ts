import { Component } from '@angular/core';

/**
 * ProgressLogItemComponent
 *
 * Type: Presentational
 *
 * Scope:
 * - Responsible for rendering an individual Progress Log Item.
 * - Manages the styling and layout necessary for the item's presentation.
 *
 * Out-of-Scope:
 * - Should not handle data fetching.
 * - Contains no business logic related to the item.
 * - Does not manage or transform the item's data.
 *
 * Purpose (optional):
 * To visually present an individual progress log item.
 */
@Component({
    selector: 'app-progress-log-item',
    imports: [],
    templateUrl: './progress-log-item.component.html',
})
export class ProgressLogItemComponent {}
