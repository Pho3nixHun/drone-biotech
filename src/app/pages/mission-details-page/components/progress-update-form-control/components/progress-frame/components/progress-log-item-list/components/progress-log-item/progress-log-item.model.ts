import { isObject } from '@utils/is-object.typeguard';
import { ProgressItemType } from 'src/app/pages/mission-details-page/mission-details-page.model';

export interface ProgressLogItemVM {
    statusType: ProgressItemType;
    date: Date;
    description: string;
}

export const isProgressLogItemVM = (obj: unknown): obj is ProgressLogItemVM =>
    isObject(obj) &&
    'statusType' in obj &&
    'date' in obj &&
    'description' in obj;
