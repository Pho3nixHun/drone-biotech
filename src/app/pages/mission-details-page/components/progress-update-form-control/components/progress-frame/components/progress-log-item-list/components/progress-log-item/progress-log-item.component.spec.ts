import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ProgressLogItemComponent } from './progress-log-item.component';
import { Component } from '@angular/core';

@Component({
    imports: [ProgressLogItemComponent],
    template: `
        <app-progress-log-item>
            <div>Should not be projected</div>
            <h3>Should be projected</h3>
            <p>Should be projected</p>
            <time datetime="">Should be projected</time>
        </app-progress-log-item>
    `,
})
class TestHostComponent {}

describe('ProgressLogItemComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should project the content and render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
