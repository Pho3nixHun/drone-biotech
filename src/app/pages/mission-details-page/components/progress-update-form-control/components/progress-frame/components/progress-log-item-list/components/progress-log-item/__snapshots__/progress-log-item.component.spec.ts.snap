// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressLogItemComponent should project the content and render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-progress-log-item>
    <div
      class="flex flex-row justify-between items-center px-2 py-1 space-x-4 w-full"
    >
      <div
        class="break-all"
      >
        <div
          class="font-medium text-sm"
        >
          <h3>
            Should be projected
          </h3>
        </div>
        <div
          class="text-gray-600 font-light text-sm"
        >
          <p>
            Should be projected
          </p>
        </div>
      </div>
      <div
        class="text-gray-600 font-light text-sm"
      >
        <time
          datetime=""
        >
          Should be projected
        </time>
      </div>
    </div>
  </app-progress-log-item>
</div>
`;
